jerry-core/ecma/base/ecma-globals.h

```diff
@@ -846,18 +857,21 @@ typedef struct
      */
     struct
     {
-      uint8_t type; /**< pseudo array type, e.g. Arguments, TypedArray*/
+      uint8_t type; /**< pseudo array type, e.g. Arguments, TypedArray, ArrayIterator */
       uint8_t extra_info; /**< extra information about the object.
-                           *   e.g. element_width_shift for typed arrays */
+                           *   e.g. element_width_shift for typed arrays,
+                           *        [[IterationKind]] property for %Iterator% */
       union
       {
         uint16_t length; /**< for arguments: length of names */
         uint16_t class_id; /**< for typedarray: the specific class name */
+        uint16_t iterator_index; /**< for %Iterator%: [[%Iterator%NextIndex]] property */
```

Comment:
Since it's part of a union it won't. The new member only represents the correct naming for the new type instead of using the already existing ones. Also we have a static assert for the size of an `ecma_object_t` which limits the size of this structure.

Reply:
@rerobika my bad, I missed that it is inside the union.